WordPress/Classes/Utility/ContextManager+ErrorHandling.swift

```diff
@@ -0,0 +1,109 @@
+    NSValidationInvalidURIError: "Some URI value cannot be represented as a string",
+    NSValidationMissingMandatoryPropertyError: "Non-optional property with a nil value",
+    NSValidationMultipleErrorsError: "Generic message for error containing multiple validation errors",
+    NSValidationNumberTooLargeError: "Some numerical value is too large",
+    NSValidationNumberTooSmallError: "Some numerical value is too small",
+    NSValidationRelationshipDeniedDeleteError: "Some relationship with NSDeleteRuleDeny is non-empty",
+    NSValidationRelationshipExceedsMaximumCountError: "Bounded, to-many relationship with too many destination objects",
+    NSValidationRelationshipLacksMinimumCountError: "To-many relationship with too few destination objects",
+    NSValidationStringPatternMatchingError: "Some string value fails to match some pattern",
+    NSValidationStringTooLongError: "Some string value is too long",
+    NSValidationStringTooShortError: "Some string value is too short",
+]
+
+private extension NSExceptionName {
+    static let coreDataSaveMainException = NSExceptionName("Unresolved Core Data save error (Main Context)")
+    static let coreDataSaveDerivedException = NSExceptionName("Unresolved Core Data save error (Derived Context)")
+}
+
+extension ContextManager {
+    @objc(handleSaveError:inContext:)
+    func handleSaveError(_ error: NSError, in context: NSManagedObjectContext) {
+        let isMainContext = context == mainContext
+        let exceptionName: NSExceptionName = isMainContext ? .coreDataSaveMainException : .coreDataSaveDerivedException
+        let reason = reasonForError(error)
+        DDLogError("Unresolved Core Data save error: \(error)")
+        DDLogError("Generating exception with reason:\n\(reason)")
+        // Sentry is choking when userInfo is too big and not sending crash reports
+        // For debugging we can still see the userInfo details since we're logging the full error above
+        let exception = NSException(name: exceptionName, reason: reason, userInfo: nil)
+        exception.raise()
+    }
+}
+
+private extension ContextManager {
+    func reasonForError(_ error: NSError) -> String {
+        if error.code == NSValidationMultipleErrorsError {
+            guard let errors = error.userInfo[NSDetailedErrorsKey] as? [NSError] else {
+                return "Multiple errors without details"
+            }
+            return reasonForMultipleErrors(errors)
+        } else {
+            return reasonForIndividualError(error)
+        }
+    }
+
+    func reasonForMultipleErrors(_ errors: [NSError]) -> String {
+        return "Multiple errors:\n" + errors.enumerated().map({ (index, error) in
+            return "  \(index + 1): " + reasonForIndividualError(error)
+        }).joined(separator: "\n")
+    }
+
+    func reasonForIndividualError(_ error: NSError) -> String {
+        let entity = entityName(for: error) ?? "null"
+        let property = propertyName(for: error) ?? "null"
```

Comment:
I think the nicer thing would be to skip them in the message if `null`, but since I don't fully understand all the possible combinations, I prefer to leave it like that for now and see what scenarios come up in crashes before trying to improve that behavior